### DISCUSSION TITLE
Are Supabase hooks secure?

### DISCUSSION THREAD

**Discussion by Nick-Mazuk** [supabase/supabase#2651] [Questions] [answered]
Are the new Supabase hooks secure? More specifically:

1. Are the HTTP messages encrypted with HTTPS (or some other encryption scheme)?
2. Is there a way to prevent malicious attackers from mimicking the webhooks?

I look forward to trying them out, but I want to be sure that any API endpoint I setup to process the webhooks can be sure that the HTTP requests originated from the database.

**Comment by inian** [comment, marked as answer]
Hi @Nick-Mazuk, 
1. If you give a webhook with HTTP protocol, we ping via HTTP and if you give a urls which is HTTPS, we use HTTPS
![CleanShot 2021-08-02 at 13 00 46](https://user-images.githubusercontent.com/2155545/127806524-f0aeb111-1247-461e-a74d-4b40428263d3.png)

2. The webhook endpoint is stored in your own database in a separate schema called `supabase_functions` which is not publicly accessible. So they are as safe as they can be.
